packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -5,151 +5,294 @@
-                              markerId:(NSString *)markerId
+                      markerIdentifier:(NSString *)identifier
                                mapView:(GMSMapView *)mapView {
   self = [super init];
   if (self) {
     _marker = [GMSMarker markerWithPosition:position];
     _mapView = mapView;
-    _markerId = markerId;
-    _marker.userData = @[ _markerId ];
-    _consumeTapEvents = NO;
+    _marker.userData = @[ identifier ];
   }
   return self;
 }
 - (void)showInfoWindow {
-  _mapView.selectedMarker = _marker;
+  self.mapView.selectedMarker = self.marker;
 }
 - (void)hideInfoWindow {
-  if (_mapView.selectedMarker == _marker) {
-    _mapView.selectedMarker = nil;
+  if (self.mapView.selectedMarker == self.marker) {
+    self.mapView.selectedMarker = nil;
   }
 }
 - (BOOL)isInfoWindowShown {
-  return _mapView.selectedMarker == _marker;
+  return self.mapView.selectedMarker == self.marker;
 }
 - (BOOL)consumeTapEvents {
-  return _consumeTapEvents;
+  return self.consumeTapEvents;
 }
 - (void)removeMarker {
-  _marker.map = nil;
+  self.marker.map = nil;
 }
 
 #pragma mark - FLTGoogleMapMarkerOptionsSink methods
 
 - (void)setAlpha:(float)alpha {
-  _marker.opacity = alpha;
+  self.marker.opacity = alpha;
 }
 - (void)setAnchor:(CGPoint)anchor {
-  _marker.groundAnchor = anchor;
+  self.marker.groundAnchor = anchor;
 }
 - (void)setConsumeTapEvents:(BOOL)consumes {
-  _consumeTapEvents = consumes;
+  self.consumeTapEvents = consumes;
 }
 - (void)setDraggable:(BOOL)draggable {
-  _marker.draggable = draggable;
+  self.marker.draggable = draggable;
 }
 - (void)setFlat:(BOOL)flat {
-  _marker.flat = flat;
+  self.marker.flat = flat;
 }
 - (void)setIcon:(UIImage *)icon {
-  _marker.icon = icon;
+  self.marker.icon = icon;
 }
 - (void)setInfoWindowAnchor:(CGPoint)anchor {
-  _marker.infoWindowAnchor = anchor;
+  self.marker.infoWindowAnchor = anchor;
 }
 - (void)setInfoWindowTitle:(NSString *)title snippet:(NSString *)snippet {
-  _marker.title = title;
-  _marker.snippet = snippet;
+  self.marker.title = title;
+  self.marker.snippet = snippet;
 }
 - (void)setPosition:(CLLocationCoordinate2D)position {
-  _marker.position = position;
+  self.marker.position = position;
 }
 - (void)setRotation:(CLLocationDegrees)rotation {
-  _marker.rotation = rotation;
+  self.marker.rotation = rotation;
 }
 - (void)setVisible:(BOOL)visible {
-  _marker.map = visible ? _mapView : nil;
+  self.marker.map = visible ? self.mapView : nil;
 }
 - (void)setZIndex:(int)zIndex {
-  _marker.zIndex = zIndex;
+  self.marker.zIndex = zIndex;
 }
+
 @end
 
-static double ToDouble(NSNumber *data) { return [FLTGoogleMapJsonConversions toDouble:data]; }
+@interface FLTMarkersController ()
 
-static float ToFloat(NSNumber *data) { return [FLTGoogleMapJsonConversions toFloat:data]; }
+@property(strong, nonatomic) NSMutableDictionary *markerIdentifierToController;
+@property(weak, nonatomic) FlutterMethodChannel *methodChannel;
```

Comment:
Why is this weak? That seems surprising.

Reply:
Bump, should this be weak?